logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -125,7 +125,7 @@ public synchronized void writeEntry(DLQEntry entry) throws IOException {
     public synchronized void writeEntry(Event event, String pluginName, String pluginId, String reason) throws IOException {
         Timestamp entryTimestamp = Timestamp.now();
         if (entryTimestamp.getTime().isBefore(lastEntryTimestamp.getTime())) {
-            entryTimestamp = lastEntryTimestamp;
+            throw new IllegalStateException("Detected concurrency issue in DLQ Writer Access.\n");
```

Comment:
Is this true? Isn't Timestamp.now subject to clock variations, and not guaranteed to be monotonic?

Reply:
@robbavey yikes right ... sec reverting thought we were on nanotime here but obviously not :)

Reply:
@original-brownbear Sadly nanotime is only used for elapsed time, not for system or wall-clock time :(

Reply:
@robbavey yea ... there wasn't a good reason to assume nano time really :D